### ISSUE TITLE
Store application name as metadata in the entity-alias

### ISSUE THREAD

**Issue by AP-Hunt** [hashicorp/vault-plugin-auth-cf#36]
Hi team,

As per #34, #35, [we're](https://cloud.service.gov.uk) looking at adopting Vault as a multi-tenant secret store for our CloudFoundry deployment.

At [`path_login.go#187`](https://github.com/hashicorp/vault-plugin-auth-cf/blob/master/path_login.go#L187), the org, space, and app guids are stored as metadata on the entity-alias which is created when an application authenticates. This is very useful because it allows us to write templated policies which restrict the app to seeing only its own secrets.

Some of our tenants perform blue-green deployments by creating a new app, renaming the old app, renaming the new app, and re-mapping the routes to the new app. This results in a new app guid. If we were restricting what apps could see using the app guid in a templated policy, the secrets would need to be set again at a new path including the new app guid. This isn't terribly ergonomic for our tenants. 

Instead, we'd like to include the app name in the metadata. That would allow us to write a policy like the below, and sidestep the issue above.

```
path "/secret-engine/{{identity.entity.aliases.cf_auth.metadata.org_id}}/{{identity.entity.aliases.cf_auth.metadata.space_id}}/{{identity.entity.aliases.cf_auth.metadata.app_name}}" {
  capability = ["read", "list"]
}
```

Is this a feature you'd be interested in incorporating? Would you accept a PR for it?

Regards,
Andy

**Comment by schmie** [comment]
Hi team,

Just a quick message to make sure the PR [#39](https://github.com/hashicorp/vault-plugin-auth-cf/pull/39) is on your radar. Not to stress, but a quick ack would be great! 

Many thanks and kind regards,

Sebastian

**Comment by barsutka** [comment]
Hi team,
Just checking if there is any estimation when the PR #39 will be reviewed?

Thanks,
Maxim.

**Comment by jasonodonnell** [comment]
Merged #39, closing!
